guppylang/std/err.py

```diff
@@ -0,0 +1,70 @@
+"""Error handling with the `Result` type."""
+
+from typing import Generic, no_type_check
+
+from guppylang_internals.decorator import custom_function, custom_type
+from guppylang_internals.definition.custom import NoopCompiler
+from guppylang_internals.std._internal.compiler.either import (
+    EitherConstructor,
+    EitherTestCompiler,
+    EitherUnwrapCompiler,
+    either_to_hugr,
+)
+from guppylang_internals.tys.param import TypeParam
+
+from guppylang import guppy
+from guppylang.std.builtins import owned
+from guppylang.std.either import Either
+
+T = guppy.type_var("T", copyable=False, droppable=False)
+E = guppy.type_var("E", copyable=False, droppable=False)
+
+_params = [TypeParam(0, "T", False, False), TypeParam(1, "E", False, False)]
+
+
+@custom_type(either_to_hugr, params=_params)
```

Comment:
why custom rather than just wrap an either value? Smaller codegen?

Reply:
this is very neat though

Reply:
> Smaller codegen?

Yes, the produced Hugr will be nicer imo. Also less code since we can reuse the `Either` compilers one-to-one